src/app/portlets/dot-edit-page/layout/dot-edit-layout-grid/dot-edit-layout-grid.component.ts

```diff
@@ -146,4 +175,52 @@ export class DotEditLayoutGridComponent implements OnInit {
         }
         return rowArray;
     }
+
+    private transformDataToDisplayOnGrid(resp): any {
+        let grid = [];
+        resp.layout.body.rows.forEach((row, rowIndex) => {
+            row.columns.forEach(column => {
+                grid.push({
```

Comment:
Can you just do this with ```Array.prototype.map``` instead of ```.push```?

Reply:
I am creating a hole new object, so push is needed here.